### DISCUSSION TITLE
Question for horizontal label position or composition pattern for input wrapper components.

### DISCUSSION THREAD

**Discussion by casamia918** [mantinedev/mantine#6000] [Feature requests]
## 1. Reference
https://www.w3.org/WAI/WCAG21/Techniques/general/G162

## 2. Current status
- Mantine supports customization of input component by [Input.Wrapper](https://mantine.dev/core/input/#inputwrapper-component)
- But, it only support label, description and error as props, not as compositional ways
    - it has inputWrapperOrder props, but it is not sufficient for displaying "Horizontal label", because all the "contents" are reside in single Box element.
- [InputWrapper SourceCode](https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/Input/InputWrapper/InputWrapper.tsx)

## 3. Requirement or Question
- Does Mantine devteam has plans for support of horizontal label position, or compositional ways to make input wrapper component? 
    - Horizontal label position is well described on w3 standard spec.
    - If composition pattern is supported, developers can get freedom to manipulate dom structure of "label, input, description, error" portions. So no need to support horizontal because it is developer user's own responsibility.
- Can I make PR for horizontal label position or should I just wait for the devteam's decision? 
    - I think this is sensitive issue for input component. So I want to get answers before I make PR code

## 4. Description
- In these days, many ui libraries makes UI Components within Mobile first strategy. 
- But still, many applications (especially for enterprise apps) are works only on desktop browser and form inputs and input components are most important things in UI components.
- Also, the label position is important interests in UI design.

## 5. History
- At first, I tried to make my own InputWrapper component
- But I realized that [InputBase](https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/InputBase/InputBase.tsx) is based on default InputWrapper and all other input components are based on InputBase component.

**Comment by rtivital** [comment]
Horizontal labels are not planned to be added, it will make maintanance much harder.
